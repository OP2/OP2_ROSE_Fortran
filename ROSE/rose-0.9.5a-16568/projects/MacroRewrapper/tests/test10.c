
typedef struct {
  char *buffer;
  int len;
  int allocatedLen;
//  char staticBuf[DBUF_STATIC_SIZE];
} DynamicBuffer;

int DBufPutcFN(DynamicBuffer *dbuf, char c);

#define DBufPutc(dbuf, c) ( (dbuf)->allocatedLen < (dbuf)->len+1 ) ? (dbuf)->buffer[(dbuf)->len++] = c, (dbuf)->buffer[(dbuf)->len] = 0, 0 : DBufPutcFN((dbuf), c)

int main()
{
  char const **in;
  DynamicBuffer* buf;
  DBufPutc(buf, '\v');
  DBufPutc(buf, **in);

};
