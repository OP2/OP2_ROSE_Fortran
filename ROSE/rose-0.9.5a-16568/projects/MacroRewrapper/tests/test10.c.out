
Macro def at: test10.c l 11 c 9
FORMAL BODY: ( (dbuf)->allocatedLen < (dbuf)->len+1 ) ? (dbuf)->buffer[(dbuf)->len++] = c, (dbuf)->buffer[(dbuf)->len] = 0, 0 : DBufPutcFN((dbuf), c)
It has 2 calls to it

Macro Call at test10.c l 17 c 3EXPANDED: ( (buf)->allocatedLen < (buf)->len+1 ) ? (buf)->buffer[(buf)->len++] = '\v', (buf)->buffer[(buf)->len] = 0, 0 : DBufPutcFN((buf), '\v')
Matching AST: ((buf -> allocatedLen) <((buf -> len) + 1))?((((((buf -> buffer)[buf -> len++] =('\v')) ,((buf -> buffer)[buf -> len] =(0)))) , 0)) : DBufPutcFN(buf,('\v'));
 Corresponding Ast Nodes: SgVarRefExp SgVarRefExp SgIntVal SgAddOp SgLessThanOp SgVarRefExp SgVarRefExp SgPlusPlusOp SgAssignOp SgVarRefExp SgVarRefExp SgIntVal SgAssignOp SgCommaOpExp SgIntVal SgCommaOpExp SgConditionalExp 
Is inconsistent with:
Macro Call at test10.c l 18 c 3EXPANDED: ( (buf)->allocatedLen < (buf)->len+1 ) ? (buf)->buffer[(buf)->len++] = **in, (buf)->buffer[(buf)->len] = 0, 0 : DBufPutcFN((buf), **in)
Matching AST: ((buf -> allocatedLen) <((buf -> len) + 1))?((((((buf -> buffer)[buf -> len++] = *( *in)) ,((buf -> buffer)[buf -> len] =(0)))) , 0)) : DBufPutcFN(buf, *( *in));
 Corresponding Ast Nodes: SgVarRefExp SgVarRefExp SgIntVal SgAddOp SgLessThanOp SgVarRefExp SgVarRefExp SgPlusPlusOp SgAssignOp SgVarRefExp SgVarRefExp SgIntVal SgAssignOp SgCommaOpExp SgIntVal SgCommaOpExp SgConditionalExp 
Is inconsistent with:
